packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -0,0 +1,304 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+
+import '../auto_dispose.dart';
+import 'auto_dispose_mixin.dart';
+import 'common_widgets.dart';
+import 'controllers.dart';
+import 'screen.dart';
+import 'theme.dart';
+import 'utils.dart';
+
+const _runInProfileModeDocsUrl =
+    'https://flutter.dev/docs/testing/ui-performance#run-in-profile-mode';
+
+const _profileGranularityDocsUrl =
+    'https://flutter.dev/docs/development/tools/devtools/performance#profile-granularity';
+
+class BannerMessagesController implements DisposableController {
+  final _dismissedMessageIds = <String>{};
+
+  final _refreshMessagesController = StreamController<Null>.broadcast();
+
+  Stream<Null> get onRefreshMessages => _refreshMessagesController.stream;
+
+  @override
+  void dispose() {
+    _refreshMessagesController.close();
+  }
+
+  void refreshMessages() {
+    _refreshMessagesController.add(null);
```

Comment:
the body of this method doesn't seem related to refreshing. can we make this a bit more idiomatic?